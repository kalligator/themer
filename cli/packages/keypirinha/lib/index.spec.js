const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('themer keypirinha theme generator', () => {
  const promisedFiles = Promise.all(render(colors));
  const colorPropertyNames = [
    'color_background',
    'color_foreground',
    'color_faded',
    'color_accent',
    'color_warn',
    'color_title',
    'color_status',
    'color_textbox_back',
    'color_listitem_back',
    'color_listitem_title',
    'color_listitem_desc',
    'color_listitem_tips',
    'color_listitem_selected_back',
    'color_listitem_selected_title',
    'color_listitem_selected_desc',
    'color_listitem_selected_tips',
  ];

  it('should generate well-formatted themes', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
    files.forEach((file) => {
      expect(/themer-keypirinha-(dark|light)\.ini/.test(file.name)).toBe(true);
      fileLines = file.contents.toString('utf8').split('\n');
      expect(/\[theme\/Themer(Dark|Light)Colors\]/.test(fileLines[0])).toBe(
        true,
      );
      for (let i = 1; i < fileLines.length; i++) {
        if (fileLines[i] === '') continue;
        else {
          [colorPropertyName, color] = fileLines[i].split('=');
          expect(
            colorPropertyNames.indexOf(colorPropertyName.trim()),
          ).toBeGreaterThan(-1);
          expect(/^#[a-fA-F0-9]+$/.test(color.trim())).toBe(true);
        }
      }
    });
  });

  it('should provide installation instructions', async () => {
    const files = await promisedFiles;
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();

    const singleThemeInstructions = renderInstructions([files[1].name]);
    expect(singleThemeInstructions).toMatchSnapshot();
  });
});
