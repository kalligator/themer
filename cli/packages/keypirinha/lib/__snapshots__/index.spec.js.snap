// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer keypirinha theme generator should provide installation instructions 1`] = `
"
1. Open the Keypirinha Configuration:
    - Select **Keypirinha: Configure** item from the Keypirinha LaunchBox
    - Right-Click **Keypirinha** -> **Configure Keypirinha**
2. Add the contents of 'themer-keypirinha-dark.ini' and 'themer-keypirinha-light.ini' to the \`Keypirinha.ini\` file.
3. Set/Append the desired theme name (\\"ThemerDarkColors\\" or \\"ThemerLightColors\\") to the \`theme\` property under the \`[gui]\` section of the \`Keypirinha.ini\` file.
"
`;

exports[`themer keypirinha theme generator should provide installation instructions 2`] = `
"
1. Open the Keypirinha Configuration:
    - Select **Keypirinha: Configure** item from the Keypirinha LaunchBox
    - Right-Click **Keypirinha** -> **Configure Keypirinha**
2. Add the contents of 'themer-keypirinha-light.ini' to the \`Keypirinha.ini\` file.
3. Set/Append the desired theme name (\\"ThemerLightColors\\") to the \`theme\` property under the \`[gui]\` section of the \`Keypirinha.ini\` file.
"
`;
